src/modules/LessonsWithFilter/LessonsWithFilter.tsx

```diff
@@ -80,6 +84,9 @@ export const LessonsWithFilter = () => {
           <Dropdown
             options={subjectOptions}
             onOptionSelect={(value: string) => setFilter('subjectIds', value)}
+            selectedOption={
+              filter.subjectIds ? filter.subjectIds[0].toString() : undefined
```

Comment:
This ternary condition actually checks this: `filter.subjectsIds !== undefined`. When `subjectIds` is empty array (`[]`), this condition: `[] !== undefined` evaluates to `true` as empty array is "something", not undefined. Then `subjectIds[0]` returns `undefined` as the index targets non-existing value. Calling `toString()` will throw if I'm not mistaken.

Reply:
It should be like `filter.subjectIds?.length !== 0`

Reply:
That shouldn;t work too, I think. Because `undefined !== 0` is `true`.

This should do it: `filter.subjectIds && filter.subjectIds.length !== 0`. Ugly, but should work I think.

Reply:
Yeah that will works. Sorry, I'm on Sprint planning at work and looking at this simultaneously so I'm writing garbage. :)